src/Propel/Generator/Builder/Om/ObjectBuilder.php

```diff
@@ -1804,11 +1881,19 @@ protected function addJsonMutator(&$script, Column $col)
     {
         $clo = $col->getLowercasedName();
 
-        $this->addMutatorOpen($script, $col);
+        $this->addJsonMutatorOpen($script, $col);
 
         $script .= "
-        \$this->$clo = json_encode(\$v);
-        \$this->modifiedColumns[".$this->getColumnConstant($col)."] = true;
+        if (is_array(\$v)) {
+            \$v = json_decode(json_encode(\$v));
```

Comment:
I don't understand. Why don't compare the actual object, but the json-string of it. Where is a object comparison happen?